File: dashboard/src/modules/trace-explorer/interfaces/Trace.ts
import type { TraceId } from '../ids';
import type { UserId } from '../../../common/ids';

export interface Trace {
  trace_id: TraceId;
  name: string;
  start_time_ms: number;
  end_time_ms: number;
  duration_ms: number;
  user_id: UserId;
  span_count: number;
  has_error: boolean;
  total_cost_usd: number;
}

File: api/internal/model/trace.go
package model

// Trace represents a summarised view of a distributed trace.
type Trace struct {
	TraceID      string  `json:"trace_id"`
	Name         string  `json:"name"`
	StartTimeMs  int64   `json:"start_time_ms"`
	EndTimeMs    int64   `json:"end_time_ms"`
	DurationMs   int64   `json:"duration_ms"`
	TenantID     string  `json:"tenant_id"`
	SpanCount    int     `json:"span_count"`
	HasError     bool    `json:"has_error"`
	TotalCostUSD float64 `json:"total_cost_usd"`
}

// TraceDetail extends Trace with the full list of constituent spans.
type TraceDetail struct {
	Trace
	Spans []Span `json:"spans"`
}
